moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h

```diff
@@ -949,6 +952,10 @@ class PlanningScene : private boost::noncopyable, public std::enable_shared_from
 
   MOVEIT_STRUCT_FORWARD(CollisionDetector)
 
+  /* Construct a new CollisionDector from allocator, copy-construct environments from parent_detector if specified */
+  void allocateCollisionDetector(const collision_detection::CollisionDetectorAllocatorPtr& allocator,
+                                 const CollisionDetectorPtr& parent_detector);
```

Comment:
I'm not sure this `parent_detector` argument is needed. Either there is a non-null `parent_` or there isn't.